plugin-build/src/main/kotlin/io/sentry/android/gradle/telemetry/SentryTelemetryService.kt

```diff
@@ -0,0 +1,474 @@
+        hub.traceparent?.let { header ->
+            args.add("--header")
+            args.add("${header.name}:${header.value}")
+        }
+        hub.baggage?.let { header ->
+            args.add("--header")
+            args.add("${header.name}:${header.value}")
+        }
+        return args
+    }
+
+    fun startTask(operation: String): ISpan? {
+        didAddChildSpans = true
+        hub.setTag("step", operation)
+        return hub.span?.startChild(operation)
+    }
+
+    fun endTask(span: ISpan?, task: Task) {
+        span?.let { span ->
+            task.state.failure?.let { throwable ->
+                captureError(throwable, span.operation)
+                span.status = SpanStatus.UNKNOWN_ERROR
+            }
+
+            span.finish()
+        }
+    }
+
+    override fun close() {
+        if (transaction?.isFinished == false) {
+            endRun()
+        }
+        Sentry.close()
+    }
+
+    companion object {
+        val SENTRY_SAAS_DSN: String =
+            "https://dd1f82ad30a331bd7def2a0dce926c6e@o447951.ingest.sentry.io/4506031723446272"
+        val MECHANISM_TYPE: String = "GradleTelemetry"
+        private val orgRegex = Regex("""(?m)Default Organization: (.*)$""")
+
+        fun createParameters(
+            project: Project,
+            variant: SentryVariant?,
+            extension: SentryPluginExtension,
+            cliExecutable: String,
+            sentryOrg: String?,
+            buildType: String
+        ): SentryTelemetryServiceParams {
+            val tags = extraTagsFromExtension(project, extension)
+            var isSaas: Boolean? = null
+            var defaultSentryOrganization: String? = null
+
+            if (isExecAvailable()) {
+                return paramsWithExecAvailable(
+                    project,
+                    cliExecutable,
+                    extension,
+                    variant,
+                    isSaas,
+                    defaultSentryOrganization,
+                    sentryOrg,
+                    buildType,
+                    tags
+                )
+            } else {
+                return SentryTelemetryServiceParams(
+                    extension.telemetry.get(),
+                    extension.telemetryDsn.get(),
+                    sentryOrg,
+                    buildType,
+                    tags,
+                    cliVersion = BuildConfig.CliVersion
+                )
+            }
+        }
+
+        private fun paramsWithExecAvailable(
+            project: Project,
+            cliExecutable: String,
+            extension: SentryPluginExtension,
+            variant: SentryVariant?,
+            isSaas: Boolean?,
+            defaultSentryOrganization: String?,
+            sentryOrg: String?,
+            buildType: String,
+            tags: Map<String, String>
+        ): SentryTelemetryServiceParams {
+            var isSaas1 = isSaas
+            var defaultSentryOrganization1 = defaultSentryOrganization
```

Comment:
These two are actually unused, I guess you forgot to pass them into `SentryTelemetryServiceParams`?

Reply:
will check

Reply:
oops